### PR TITLE
Bug 1951456: Docs and lint fixes

### PR DESCRIPTION
**Pull request by 0sewa0**
Goes through the codebase and adds/updates the docs for public functions and fixes the linting issues

## Categories

- [ ] Bugfix
- [ ] Enhancement
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample archive

- none

## Documentation

- none

## Unit Tests

- none

## Privacy

Yes. There are no sensitive data in the newly collected information.

### CODE REVIEW

File: pkg/anonymization/anonymizer.go [hunk 1]
```diff
@@ -113,18 +112,10 @@ func NewAnonymizerFromConfigClient(
...
-	for _, network := range networksConfig.Spec.ExternalIP.AutoAssignCIDRs {
-		networks = append(networks, network)
-	}
-	for _, network := range networksConfig.Spec.ExternalIP.Policy.AllowedCIDRs {
-		networks = append(networks, network)
-	}
-	for _, network := range networksConfig.Spec.ExternalIP.Policy.RejectedCIDRs {
-		networks = append(networks, network)
-	}
+	networks = append(networks, networksConfig.Spec.ServiceNetwork...)
```
Comment (rluders) [inline pkg/anonymization/anonymizer.go:115]: :open_mouth: beaultiful.

File: pkg/config/configobserver/configobserver.go [hunk 2]
```diff
@@ -294,12 +319,28 @@ func (c *Controller) setConfigLocked(config *config.Controller) {
 			}
 		}
 	} else {
-		klog.V(2).Infof(
-			"Configuration set: enabled=%t endpoint=%s interval=%s username=%t token=%t reportEndpoint=%s initialPollingDelay=%s minRetryTime=%s pollingTimeout=%s",
-			config.Report, config.Endpoint, config.Interval, len(config.Username) > 0, len(config.Token) > 0, config.ReportEndpoint,
-			config.ReportPullingDelay, config.ReportMinRetryTime, config.ReportPullingTimeout)
+		klog.V(2).Infof( //nolint: gomnd
+			"Configuration updated: "+
```
Comment (Sergey1011010) [inline pkg/config/configobserver/configobserver.go:323]: `set` became `updated`, maybe it's better to refactor it somehow. What about creating a function `toStr` on config?

File: pkg/controller/status/status.go [hunk 3]
```diff
@@ -39,6 +39,8 @@ type Configurator interface {
 	Config() *config.Controller
 }
 
+// Responsible for managing the status of the operator according the status of the sources.
```
Comment (Sergey1011010) [inline pkg/controller/status/status.go:42]: according to

File: pkg/controller/status/status.go [hunk 4]
```diff
@@ -39,6 +39,8 @@ type Configurator interface {
...
+// Sources come from different major parts of the codebase, representing the status of said part of the codebase.
```
Comment (Sergey1011010) [inline pkg/controller/status/status.go:43]: `of said part`? What does it mean? :)

File: pkg/controller/status/status.go [hunk 5]
```diff
@@ -53,7 +55,10 @@ type Controller struct {
 	start    time.Time
 }
 
-func NewController(client configv1client.ConfigV1Interface, coreClient corev1client.CoreV1Interface, configurator Configurator, namespace string) *Controller {
+// Creates a status conteroller, responsible for monitoring the operators status and updating the it's cluster status accordingly.
```
Comment (Sergey1011010) [inline pkg/controller/status/status.go:58]: `conteroller` typo

File: pkg/controller/status/status.go (hunk 4 above)
Comment (0sewa0) [reply pkg/controller/status/status.go:43]: Its trying to reference the `major` parts in the previous part of the sentence :) Will try to make it clearer

File: pkg/config/config.go [hunk 6]
```diff
@@ -58,7 +58,31 @@ type HTTPConfig struct {
...
 type Converter func(s *Serialized, cfg *Controller) (*Controller, error)
 
-func ToController(s *Serialized, cfg *Controller) (*Controller, error) {
+// Returns the important fields of the config in a string form
+func (c *Controller) ToString() string {
+	return fmt.Sprintf("enabled=%t "+
+		"endpoint=%s "+
+		"interval=%s "+
+		"username=%t "+
+		"token=%t "+
```
Comment (tremes) [inline pkg/config/config.go:67]: I know we had it before, but should we log this? :)

File: pkg/config/config.go [hunk 7]
```diff
@@ -129,6 +153,8 @@ func ToController(s *Serialized, cfg *Controller) (*Controller, error) {
 	return cfg, nil
 }
 
+// Creates/updates a config Controller according to the Serialized config.
+// Makes sure that the config is correct, but only checks fields necessary for disconnected operatotion.
```
Comment (tremes) [inline pkg/config/config.go:157]: Typo `operatotion`

File: pkg/config/config.go (hunk 6 above)
Comment (0sewa0) [reply pkg/config/config.go:67]: nope, we probably shouldn't :D

Comment (0sewa0) [reply pkg/config/config.go:67]: but we only log if we have a token or not. (`len(c.Token) > 0`)
so its fine

File: pkg/controller/operator.go [hunk 8]
```diff
@@ -45,7 +45,7 @@ type Operator struct {
 // 3. Initiates the recorder and starts the periodic record pruneing
 // 4. Starts the periodic gathering
 // 5. Creates the insights-client and starts uploader and reporter
-func (s *Operator) Run(ctx context.Context, controller *controllercmd.ControllerContext) error {
+func (s *Operator) Run(ctx context.Context, controller *controllercmd.ControllerContext) error { //nolint: funlen
```
Comment (rluders) [inline pkg/controller/operator.go:48]: Why did you ignore the `funlen` here, but at the `status.NewController` you put it into separated lines?

Comment (0sewa0) [reply pkg/controller/operator.go:48]: here `funlen`  is about how many statements there are in the `Run` method so the `Run` would need to be chopped into smaller functions, while `status.NewController` had a really long function signature which was easy to fix :)

**Review by rluders — APPROVED** [review]
/lgtm

**Pull request merged**